test/llvm-intrinsics/uadd.with.overflow.ll

```diff
@@ -1,51 +1,170 @@
+; CHECK-SPIRV:          [[var_50:%[a-z0-9_]+]] = OpCompositeExtract [[bool]] [[var_48]] 1
+; CHECK-SPIRV:                                   OpReturnValue [[var_50]]
+; CHECK-SPIRV:                                   OpFunctionEnd
+
+; CHECK-LLVM:   entry:
+; CHECK-LLVM:   %0 = alloca [[structtype_3]], align 8
+; CHECK-LLVM:   call spir_func void @_Z17__spirv_IAddCarryll(ptr sret([[structtype_3]]) %0, i64 %a, i64 %b)
+; CHECK-LLVM:   %1 = bitcast ptr %0 to ptr
+; CHECK-LLVM:   %2 = load [[structtype_4]], ptr %1, align 4
+; CHECK-LLVM:   %3 = extractvalue [[structtype_4]] %2, 0
+; CHECK-LLVM:   %4 = extractvalue [[structtype_4]] %2, 1
+; CHECK-LLVM:   ret i1 %4
+
+define spir_func <4 x i1> @test_uadd_with_overflow_v4i32(<4 x i32> %a, <4 x i32> %b) {
 entry:
- %res = call {<4 x i32>, <4 x i1>} @llvm.uadd.with.overflow.v4i32(<4 x i32> %a, <4 x i32> %b) 
- ret void
+  %res = call {<4 x i32>, <4 x i1>} @llvm.uadd.with.overflow.v4i32(<4 x i32> %a, <4 x i32> %b) 
+  %0 = extractvalue {<4 x i32>, <4 x i1>} %res, 0
+  %1 = extractvalue {<4 x i32>, <4 x i1>} %res, 1
+  ret <4 x i1> %1
 }
 
+; CHECK-SPIRV:             [[a_2:%[a-z0-9_]+]] = OpFunctionParameter [[v4uint]]
+; CHECK-SPIRV:             [[b_2:%[a-z0-9_]+]] = OpFunctionParameter [[v4uint]]
+; CHECK-SPIRV:         [[entry_2:%[a-z0-9_]+]] = OpLabel
+; CHECK-SPIRV:          [[var_60:%[a-z0-9_]+]] = OpVariable [[_ptr_Function__struct_58]] Function
+; CHECK-SPIRV:          [[var_61:%[a-z0-9_]+]] = OpIAddCarry [[_struct_58]] [[a_2]] [[b_2]]
+; CHECK-SPIRV:                                   OpStore [[var_60]] [[var_61]]
+; CHECK-SPIRV:          [[var_64:%[a-z0-9_]+]] = OpBitcast [[_ptr_Function__struct_62]] [[var_60]]
+; CHECK-SPIRV:          [[var_65:%[a-z0-9_]+]] = OpLoad [[_struct_62]] [[var_64]] Aligned 16
+; CHECK-SPIRV:          [[var_66:%[a-z0-9_]+]] = OpCompositeExtract [[v4uint]] [[var_65]] 0
+; CHECK-SPIRV:          [[var_67:%[a-z0-9_]+]] = OpCompositeExtract [[v4bool]] [[var_65]] 1
+; CHECK-SPIRV:                                   OpReturnValue [[var_67]]
+; CHECK-SPIRV:                                   OpFunctionEnd
+
+; CHECK-LLVM:   entry:
+; CHECK-LLVM:   %0 = alloca [[structtype_5]], align 16
+; CHECK-LLVM:   call spir_func void @_Z17__spirv_IAddCarryDv4_iS_(ptr sret([[structtype_5]]) %0, <4 x i32> %a, <4 x i32> %b)
+; CHECK-LLVM:   %1 = bitcast ptr %0 to ptr
+; CHECK-LLVM:   %2 = load [[structtype_6]], ptr %1, align 16
+; CHECK-LLVM:   %3 = extractvalue [[structtype_6]] %2, 0
+; CHECK-LLVM:   %4 = extractvalue [[structtype_6]] %2, 1
+; CHECK-LLVM:   ret <4 x i1> %4
 declare {i16, i1} @llvm.uadd.with.overflow.i16(i16 %a, i16 %b)
 declare {i32, i1} @llvm.uadd.with.overflow.i32(i32 %a, i32 %b)
 declare {i64, i1} @llvm.uadd.with.overflow.i64(i64 %a, i64 %b)
 declare {<4 x i32>, <4 x i1>} @llvm.uadd.with.overflow.v4i32(<4 x i32> %a, <4 x i32> %b)
+declare void @_Z17__spirv_IAddCarryii(ptr sret({i32, i32}), i32, i32)
```

Comment:
Why this should be declared? I can't see it's used anywhere in the test. Same for usub test.

Reply:
In regularization procedure we have two branches which condition on the existence of the function.
This line is added to cover one of the branches -  situation when somewhere in code direct builtin with __spirv_IAddCarryii is used. 
The line: 
```
  Function *Builtin = Call->getModule()->getFunction(MangledName);
  AllocaInst *A;
  if (Builtin) {
```
to trigger `true` branch.

Reply:
Okay, just a bit strange that if we remove the declaration the test will fail (or not?).
Maybe we need to have tests to cover both if-branches.